pallet/src/lib.rs

```diff
@@ -552,14 +534,54 @@ pub mod pallet {
             Ok(accounts)
         }
 
-        fn chore_multisig_storage(key: CallHash, block_tbr: BlockNumberFor<T>) -> Option<CallHash> {
-            let multisig = MultisigStorage::<T>::get(key)?;
-            if *multisig.block_number() > block_tbr {
-                None
-            } else {
-                MultisigStorage::<T>::remove(key);
-                Some(key)
+        fn new_multi_sign_request_chore(
+            block_height: BlockNumberFor<T>,
+            hash: CallHash,
+        ) -> Result<(), Error<T>> {
+            // Increase the actual block-height by the maximum lifetime of a request.
+            let block_height = block_height + T::MaxLifetimeRequests::get();
+
+            // Check for an existing entry or create an empty one.
+            let mut hashes_ready_for_cleanup: BoundedVec<CallHash, MaxChoreEntriesPerVec> =
+                ChoreOnIdleStorage::<T>::try_get(block_height).unwrap_or_default();
+
+            // Check if this hash is already included, then it is ok.
+            for h in hashes_ready_for_cleanup.iter() {
+                if *h == hash {
+                    return Ok(());
+                }
+            }
```

Comment:
If we could use `BoundedBTreeMap` within the `BoundedBTreeMap` instead of `BoundedVec`, we could avoid this search. Not sure, something to think about

Reply:
Would be a `BoundedBTreeMap<CallHash, (), Size>` because we do not store anything more than the hash value.

Reply:
After a Slack discussion, we decided to keep the current solution since we would need basically a `HashSet` here instead